src/Logary/Configuration_Config.fs

```diff
@@ -104,6 +105,14 @@ let withMetrics ms conf =
   |> Seq.map Metric.validate
   |> Seq.fold (fun s t -> s |> withMetric t) conf
 
+[<CompiledName "WithMiddleware">]
+let withMiddleware middleware conf =
+  { conf with LogaryConf.middleware = middleware :: conf.middleware }
```

Comment:
Middleware registered first will run first. So

``` fsharp
|> withMiddleware messageId
|> withMiddleware principalDetails
```

will first run messageId, then `principalDetails`. So if `messageId` is the same middleware as we have in logary-js, then this order would be in error.